core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -79,6 +79,16 @@ public final Query toQuery(QueryParseContext parseContext) throws IOException {
         return query;
     }
 
+    /**
+     * Few queries allow you to set the boost and queryName but don't do anything with it. This method allows
+     * to disable boost and queryName for those queries.
+     */
+    //norelease this has to be put in here because of the call on toQuery which would override the boost and queryName
+    //norelease of the wrapped query with defaults
+    protected boolean supportsBoostAndQueryName() {
```

Comment:
//norelease is used for code that can be removed later on in the refactoring. I don't think it should be used here, or at least I don't see how we can remove this method if we introduce it here. The idea for `supportsBoostAndQueryName` was to be only a test method to "mark" few exceptional queries, it wasn't meant to be moved to prod code ever.

The reason why you did this is because the wrapped query has its own boost that gets overwritten by the default 1.0f boost of the outer query. I tend to think that this is more a test problem, and the doAssertLuceneQuery in WrapperQueryBuilderTest should be adjusted. Maybe there are other ways or ideas, but we should not move this `supportsBoostAndQueryName` to prod code I think.